telemetry/record_test.go

```diff
@@ -31,4 +38,48 @@ var _ = Describe("Socket handler test", func() {
 		Expect(err).To(Not(BeNil()))
 		Expect(record.Serializer).To(Not(BeNil()))
 	})
+	It("includes vin in body", func() {
+		logger, _ := test.NewNullLogger()
+		serializer := telemetry.NewBinarySerializer(
+			&telemetry.RequestIdentity{
+				DeviceID: "42",
+				SenderID: "vehicle_device.42",
+			},
+			map[string][]telemetry.Producer{"D4": nil},
+			false,
+			logger,
+		)
+		message := messages.StreamMessage{TXID: []byte("1234"), SenderID: []byte("vehicle_device.42"), MessageTopic: []byte("V"), Payload: generatePayload("cybertruck", "42", nil)}
+		recordMsg, err := message.ToBytes()
+		Expect(err).To(BeNil())
+
+		record, err := telemetry.NewRecord(serializer, recordMsg, "1")
+		Expect(record).To(Not(BeNil()))
+		Expect(err).To(BeNil())
+		Expect(record.Serializer).To(Not(BeNil()))
+
+		data := &protos.VehicleMetrics{}
```

Comment:
why do we use a `protos.VehicleMetrics` here if we use a `protos.Payload` in generatePayload?